modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -31,7 +32,8 @@ InputParameters validParams<TensorMechanicsAction>()
 }
 
 TensorMechanicsAction::TensorMechanicsAction(const InputParameters & params) :
-    Action(params)
+    Action(params),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : "")
```

Comment:
You haven't described why we need a base name here. It is not like we will have multiple stress divergence kernel sets here, or will we?